x-pack/plugin/security/cli/src/main/java/org/elasticsearch/xpack/security/cli/AutoConfigureNode.java

```diff
@@ -942,14 +942,18 @@ private void notifyOfFailure(boolean inEnrollmentMode, Terminal terminal, Termin
         }
     }
 
-    @SuppressForbidden(reason = "Commons IO lib uses the File API")
+    @SuppressForbidden(reason = "We need to interact with the file system in order to perform auto configuration")
     private void deleteDirectory(Path directory) throws IOException {
-        FileUtils.deleteDirectory(directory.toFile());
+        Files.walk(directory)
+            .sorted(Comparator.reverseOrder()) // Ensures directory contents are deleted first
+            .map(Path::toFile)
+            .map(File::delete);
```

Comment:
I have no strong opinion, but while looking at this yesterday I noticed we also have:
- https://github.com/elastic/elasticsearch/blob/64798df2f344597767a675a71ba5ec6ddb7aa6ce/libs/core/src/main/java/org/elasticsearch/core/IOUtils.java#L196
and
- https://github.com/elastic/elasticsearch/blob/b2c9028384253416bd3e17de9e45af5eefe32764/server/src/main/java/org/elasticsearch/common/io/FileSystemUtils.java#L85

Perhaps one of them would do the job (or part of the job) here.

Reply:
++ reusing `IOUtils#rm` sounds like a good idea.
I struggle to remember why I didn't do that in the first place.